mlir/test/python/ir/auto_location.py

```diff
@@ -0,0 +1,101 @@
+# RUN: %PYTHON %s | FileCheck %s
+# REQUIRES: python-ge-311
+import gc
+from contextlib import contextmanager
+
+from mlir.ir import *
+from mlir.dialects._ods_common import _cext
+from mlir.dialects import arith, _arith_ops_gen
+
+
+def run(f):
+    print("\nTEST:", f.__name__)
+    f()
+    gc.collect()
+    # assert Context._get_live_count() == 0
```

Comment:
Delete line?

Reply:
err whoops this needs to be uncommented (man i hope this doesn't break the test 🤦)

Reply:
😬

Reply:
man what an ordeal - executive summary here https://github.com/jax-ml/jax/issues/30930